KubeArmor/core/kubeArmor.go

```diff
@@ -694,22 +696,46 @@ func KubeArmor() {
 	// == //
 
 	if dm.K8sEnabled && cfg.GlobalCfg.Policy {
-		// watch k8s pods
-		go dm.WatchK8sPods()
-		dm.Logger.Print("Started to monitor Pod events")
+		timeout, err := time.ParseDuration(cfg.GlobalCfg.InitTimeout)
+		if err != nil {
+			dm.Logger.Warnf("Not a valid InitTimeout duration: %q, defaulting to '60s'", cfg.GlobalCfg.InitTimeout)
+			timeout = 60 * time.Second
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
 
 		// watch security policies
-		go dm.WatchSecurityPolicies()
+		securityPoliciesSynced := dm.WatchSecurityPolicies()
+		if securityPoliciesSynced == nil {
+			return
+		}
 		dm.Logger.Print("Started to monitor security policies")
 
 		// watch default posture
-		go dm.WatchDefaultPosture()
+		defaultPostureSynced := dm.WatchDefaultPosture()
+		if defaultPostureSynced == nil {
+			return
```

Comment:
Let's have a proper cleanup here and at all new returns by calling the `DestroyKubeArmorDaemon()` function.
Example - https://github.com/kubearmor/KubeArmor/pull/1655/files#diff-fa91eb0e35e389b4f70187ae5fd27e9608b1dc719b240edb96cc6dea3fc41deeR683